prime-router/src/testIntegration/kotlin/transport/SftpTransportIntegrationTests.kt

```diff
@@ -456,10 +457,11 @@ class SftpTransportIntegrationTests : TransportIntegrationTests() {
     @Test
     fun `ls happy path`() {
         val f = Fixture()
+        val nullRemoteResourceFilter: RemoteResourceFilter? = null
 
         every { f.mockSSHClient.newSFTPClient() } returns f.mockSFTPClient
         // mock a successful ls on the remote SFTP server
-        every { f.mockSFTPClient.ls(f.lsPath, null) } returns f.remoteResourceInfos
+        every { f.mockSFTPClient.ls(f.lsPath, nullRemoteResourceFilter) } returns f.remoteResourceInfos
```

Comment:
SSHJ 0.39.0 adds a new overload to the `SFTPClient.ls` function, making it necessary to disambiguate when declaring this mock. There is no ambiguity in the actual usage, only the tests need to be updated. See [this comment](https://github.com/apache/nifi/pull/9266#pullrequestreview-2304158670) for details.